crates/matrix-sdk-crypto/src/olm/session.rs

```diff
@@ -181,7 +181,7 @@ impl Session {
             #[cfg(feature = "experimental-algorithms")]
             EventEncryptionAlgorithm::OlmV2Curve25519AesSha2 => OlmV2Curve25519AesSha2Content {
                 ciphertext,
-                sender_key: self.our_identity_keys.curve25519,
+                sender_key: self.device_keys.curve25519_key().unwrap(),
```

Comment:
Is this `unwrap` OK? If so, please can we have a comment explaining why?

Reply:
It's safe, if [the FIXME](https://github.com/matrix-org/matrix-rust-sdk/pull/3517#discussion_r1636123693) gets fixed.  I'll wait for @poljar's review before changing anything here.